src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnectionBuilder.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    public class HubConnectionBuilder : IHubConnectionBuilder
+    {
+        private readonly Dictionary<KeyValuePair<string, Type>, object> _settings = new Dictionary<KeyValuePair<string, Type>, object>();
+        private ConnectionFactoryDelegate _connectionFactoryDelegate;
+
+        public void ConfigureConnectionFactory(ConnectionFactoryDelegate connectionFactoryDelegate) =>
+            _connectionFactoryDelegate = connectionFactoryDelegate;
+
+        public void AddSetting<T>(string name, T value)
```

Comment:
Why isn't this object -> object.

Reply:
Because it is generic I don't need to worry that casting to the target type could fail.

Reply:
(when getting the setting)

Reply:
I'm wondering why the key isn't an object

Reply:
If it was an object then probably the only thing that is easy to use as a key is `Type`. Otherwise you would have to expose the instance you used as a key to others who would want to read it (e.g. extensions in a different assembly) or always provide helper methods. Using `Type` can be problematic if you need to store multiple values of the same type (e.g. what key would you use if you need to store multiple strings?).

Using a string is conceptually easier and if someone wants to read the the value they don't need to use the same instance of the key to read the value.

Reply:
I think type is better. We can treat this like a feature collection.

Reply:
Blah actually string is fine for now. Lets keep it for the alpha, we can always revisit.